src/test/java/mops/presentation/DirectoryControllerTest.java

```diff
@@ -0,0 +1,149 @@
+package mops.presentation;
+
+import mops.businesslogic.*;
+import mops.persistence.FileInfo;
+import mops.presentation.utils.SecurityContextUtil;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+import java.util.ArrayList;
+import java.util.Set;
+
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.mock;
+import static org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestPostProcessors.csrf;
+import static org.springframework.security.test.web.servlet.setup.SecurityMockMvcConfigurers.springSecurity;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class DirectoryControllerTest {
+    /**
+     * Necessary mock until GroupService is implemented.
+     */
+    @MockBean
+    private GroupService groupService;
+    /**
+     * Necessary mock until FileService is implemented.
+     */
+    @MockBean
+    private FileService fileService;
+    /**
+     * Necessary mock until DirectoryService is implemented.
+     */
+    @MockBean
+    private DirectoryService directoryService;
+    /**
+     * Necessary bean.
+     */
+    @Autowired
+    private WebApplicationContext context;
+
+    /**
+     * Necessary bean.
+     */
+    private MockMvc mvc;
+
+    /**
+     * Setups the a Mock MVC Builder.
+     */
+    @BeforeEach
+    public void setUp() {
+        ArrayList<FileInfo> files = new ArrayList<>();
```

Comment:
please use `List.of()` on both mock configs instead.

Reply:
I don't see the advantage? It is irrelevant what is returned there as long as something is returned.

Reply:
One should not use `ArrayList<>` as a variable type.
And `List.of()` is the preferred way for empty immutable lists.